runtime/src/main/cpp/Worker.cpp

```diff
@@ -494,42 +518,41 @@ void Future::cancelUnlocked() {
 // Defined in RuntimeUtils.kt.
 extern "C" void ReportUnhandledException(KRef e);
 
-void* workerRoutine(void* argument) {
-  Worker* worker = reinterpret_cast<Worker*>(argument);
-
-  g_currentWorkerId = worker->id();
-  Kotlin_initRuntimeIfNeeded();
-
-  {
-    ObjHolder argumentHolder;
-    ObjHolder resultHolder;
-    while (true) {
-      Job job = worker->getJob();
-      if (job.kind == JOB_TERMINATE) {
-        if (job.terminationRequest.waitDelayed) {
-          if (worker->waitDelayed()) {
-            worker->putJob(job, false);
-            continue;
-          }
+JobKind Worker::processQueueElement(bool blocking) {
+  ObjHolder argumentHolder;
+  ObjHolder resultHolder;
+  if (terminated_) return JOB_TERMINATE;
```

Comment:
So `Worker.processQueue()` would return `true` in this case?

Reply:
Yes, maybe contract better be improved, like return false on termination requests and here.

Reply:
Has this been fixed?

Reply:
Yeah 
```
JobKind kind = worker->processQueueElement(false);
return kind != JOB_NONE && kind != JOB_TERMINATE;
```